plan/plans/join.go

```diff
@@ -330,21 +330,14 @@ func appendRow(prefix []interface{}, in []interface{}) []interface{} {
 		t := in[len(in)-1]
 		switch vt := t.(type) {
 		case *RowKeyList:
-			if rks == nil {
-				rks = vt
-			} else {
-				rks.appendKeys(vt.Keys...)
-			}
+			rks.appendKeys(vt.Keys...)
 			in = in[:len(in)-1]
 		}
 	}
-	if rks == nil {
-		rks = &RowKeyList{}
-	}
-	in = append(in, rks)
 	row := make([]interface{}, 0, len(prefix)+len(in))
 	row = append(row, prefix...)
 	row = append(row, in...)
+	row = append(row, rks)
```

Comment:
The Do function will be abandoned. In update.go we use Next. @coocood will remove Do() later.